src/OrchardCore.Modules/OrchardCore.Contents/Liquid/DisplayUrlFilter.cs

```diff
@@ -31,12 +32,15 @@ public async ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArgument
 
             if (contentItem == null)
             {
-                if (string.IsNullOrEmpty(input.ToStringValue()))
+                if (String.IsNullOrEmpty(input.ToStringValue()))
                 {
                     return StringValue.Empty;
                 }
-                routeValues = new RouteValueDictionary(_autorouteOptions.GlobalRouteValues);
-                routeValues[_autorouteOptions.ContentItemIdKey] = input.ToStringValue();
+
+                routeValues = new RouteValueDictionary(_autorouteOptions.GlobalRouteValues)
+                {
+                    [_autorouteOptions.ContentItemIdKey] = input.ToStringValue()
```

Comment:
Okay done locally.

Didn't think about this rule because not part of our current rules, maybe would have been good to think about it, maybe another future PR to fix all of these.

Hmm, about future PRs for other good rules, maybe we could first agree on a given rule(s) and then the PR fixing a given rule would have to also integrate it in our current rules.